components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1,1802 +1,1177 @@
+  type(cam_out_t),                    dimension(begchunk:endchunk) :: cam_out
+  !-----------------------------------------------------------------------------
+  ! Local Variables
+  !-----------------------------------------------------------------------------
+  integer :: c                                 ! indices
+  integer :: ncol                              ! number of columns
+  integer :: nstep                             ! current timestep number
 #if (! defined SPMD)
-    integer :: mpicom = 0
-#endif
-    integer(i8) :: beg_count                     ! start time for a chunk
-    integer(i8) :: end_count                     ! stop time for a chunk
-    integer(i8) :: irtc_rate                     ! irtc clock rate
-    real(r8)    :: chunk_cost                    ! measured cost per chunk
-    type(physics_buffer_desc), pointer :: phys_buffer_chunk(:)
-
-    call t_startf ('physpkg_st1')
-    nstep = get_nstep()
-
-#if ( defined OFFLINE_DYN )
-    !
-    ! if offline mode set SNOWH and TS for micro-phys
-    !
-    call get_met_srf1( cam_in )
+  integer :: mpicom = 0
 #endif
+  integer(i8) :: beg_count                     ! start time for a chunk
+  integer(i8) :: end_count                     ! stop time for a chunk
+  integer(i8) :: irtc_rate                     ! irtc clock rate
+  real(r8)    :: chunk_cost                    ! measured cost per chunk
+  type(physics_buffer_desc), pointer :: phys_buffer_chunk(:)
+  !-----------------------------------------------------------------------------
+  !-----------------------------------------------------------------------------
+  call t_startf ('physpkg_st1')
+  nstep = get_nstep()
 
-    ! The following initialization depends on the import state (cam_in)
-    ! being initialized.  This isn't true when cam_init is called, so need
-    ! to postpone this initialization to here.
-    if (nstep == 0 .and. phys_do_flux_avg()) call flux_avg_init(cam_in,  pbuf2d)
+  ! The following initialization depends on the import state (cam_in)
+  ! being initialized.  This isn't true when cam_init is called, so need
+  ! to postpone this initialization to here.
+  if (nstep == 0 .and. phys_do_flux_avg()) call flux_avg_init(cam_in,  pbuf2d)
 
-    ! Compute total energy of input state and previous output state
-    call t_startf ('chk_en_gmean')
-    call check_energy_gmean(phys_state, pbuf2d, ztodt, nstep)
-    call t_stopf ('chk_en_gmean')
+  ! Compute total energy of input state and previous output state
+  call t_startf ('chk_en_gmean')
+  call check_energy_gmean(phys_state, pbuf2d, ztodt, nstep)
+  call t_stopf ('chk_en_gmean')
 
-    call t_stopf ('physpkg_st1')
+  call t_stopf ('physpkg_st1')
 
-    if ( adiabatic .or. ideal_phys )then
-       call t_startf ('bc_physics')
-       call phys_run1_adiabatic_or_ideal(ztodt, phys_state, phys_tend,  pbuf2d)
-       call t_stopf ('bc_physics')
-    else
-       call t_startf ('physpkg_st1')
+  call t_startf ('physpkg_st1')
```

Comment:
I'm not really sure what the point of this is now that you've refactored this: the `physpkg_st1` timer is stopped and then immediately restarted here. I suspect the point in the original code was to separate the `bc_physics` timer from the `physpkg_st1` timers, but now that you've moved the `bc_physics` timer below here this is probably not necessary. I think though, this has the effect of duplicating the previous behavior would which have added another call of the `physpkg_st1` timer in the timing files. I.e., doubling the number of times this timer is counted. If this is the case, I actually don't think this is a good idea, in either case (MMF or non-MMF), since it's collecting timing for different processes. I.e., the first call includes calls to `flux_avg_init` and `check_energy_gmean`, but the second just has the `pbuf_allocate`, `diag_allocate`, and `phys_timestep_init`. So I think this has the effect of doubling the number of times this timer is called, with mean values for timings being the mean of all the above processes called. Again, this appears to be how the original version behaved, but maybe we shouldn't repeat that here, as it appears to me to be a design flaw(/feature?).

Reply:
I'll just remove this timer and we can add new timers if we decide it's useful information to have.